### PR TITLE
Feature/automatedID

### PR DESCRIPTION
**Pull request by suzannemaquelin**
Id of device and activators are made upon creation, device assigns id to its activators, database assigns id to new devices

### CODE REVIEW

File: source/model/Activator.py [hunk 1]
```diff
@@ -7,8 +7,11 @@ class Activator(ABC):
     Concrete activators of a plugin should implement this class
     """
 
-    def __init__(self, id):
-        self.activatorId = id
+    @property
+    @abstractmethod
```
Comment (szijerveld) [inline source/model/Activator.py]: Just a question. As all activators require an Id I think it is best that it is not an abstract  method but that activator itself implements the id. This way a lot of duplicate code is removed.
What do you think?

File: source/model/Device.py [hunk 2]
```diff
@@ -6,16 +6,15 @@ class Device(ABC):
...
 
     def __init__(self):
-        self.deviceId = self.__class__.newId()
         self._activators = list()
 
-    @classmethod
-    def newId(cls):
-        cls.cntr += 1
-        return cls.cntr
+    @property
```
Comment (szijerveld) [inline source/model/Device.py:15]: The same as with Activator.
It would be neither if the device implements the id him self.

**Review by szijerveld — CHANGES_REQUESTED** [review]
Please have a look at my comments.
I think it will not be that much work to change it and it would make it better I think.

File: source/model/Activator.py (hunk 1 above)
Comment (suzannemaquelin) [reply source/model/Activator.py]: I agree. Changed it now

**Review by szijerveld — APPROVED** [review]

**Pull request merged**